### PR TITLE
Added an alternative DiscoverFilter constructor to allow 'OR' as an operator choice

### PR DESCRIPTION
**Pull request by Ethan-Icet**
* Added `DiscoverFilter(Separator separator, Integer... items)` to complement the existing `DiscoverFilter(Integer... items)` which uses 'AND' as default separator
  * So that 'OR' can be chosen as separator
  * Can be useful for cases when we have a list of wanted genres (keywords/crew/cast) that doesn't have all to be satisfied.
  * Example, Most popular Action, Adventure or Thriller movies:  https://api.themoviedb.org/3/discover/movie?api_key=<<api_key>>&with_genres=28|12|53&sort_by=popularity.desc
  
* Modified `test_discover_movie()` to add a with_keywords condition that covers the newly added constructor
  * The movie now have to contain either keywords {'cult film', 'insomnia', 'based on short story'}
  
* Removed some warnings in DiscoverFilter.java
  * Warnings: Dereference of 'separator' may produce 'NullPointerException'
  * Simply added some asserts
  * Could also have used some 'assertThat' but didn't knew if an additional dependency was worth it.

### CODE REVIEW

File: src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java [hunk 1]
```diff
@@ -22,7 +22,13 @@ public DiscoverFilter(Integer... items) {
         this.items = items;
     }
 
+    public DiscoverFilter(Separator separator, Integer... items) {
+        this.separator = separator;
+        this.items = items;
+    }
+
     public DiscoverFilter(Separator separator, ReleaseType... types) {
+        assert types != null;
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java:31]: Please change to a condition and throw an exception instead. Depending on the platform assert may not work. Same for the assert usage below.

File: src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java [hunk 2]
```diff
@@ -22,7 +22,13 @@ public DiscoverFilter(Integer... items) {
         this.items = items;
     }
 
+    public DiscoverFilter(Separator separator, Integer... items) {
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java:25]: Possibly change the existing constructor (`DiscoverFilter(Integer... items)`) to call this one?

**Review by UweTrottmann — CHANGES_REQUESTED** [review]
Thanks! Found some issues. Once resolved I'll merge and the changes will be available with the next update.

File: src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java (hunk 1 above)
Comment (Ethan-Icet) [reply src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java:31]: Is the following changes ok ?
```
if (types == null){
            throw new IllegalArgumentException();
}
```
```
if (separator == null){
            throw new NullPointerException();
}
```
Or are there any message conventions I should follow ?

Comment (UweTrottmann) [reply src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java]: The platform itself would throw NPE. If you really want the code to throw here explicitly, please throw `IllegalArgumentException` or `IllegalStateException` and use a message why, e.g. `types must not be null`.

Comment (Ethan-Icet) [reply src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java]: I really wanted to do a good job, if you think that's unnecessary you can ignore the last commit.
I'm not far from a beginner in Java so thanks for your feedback.

Comment (Ethan-Icet) [reply src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java]: About the `separator` being null which is not an argument in this case. I didn't find any definitive answer :
https://stackoverflow.com/questions/271526/avoiding-nullpointerexception-in-java
https://stackoverflow.com/questions/3322638/is-it-okay-to-throw-nullpointerexception-programmatically
**Annotations** beforehand in constructors, **Asserts**, **checkNotNull()**, **NPE**, **nothing**...
 Choose your school of though I will comply.

**Comment by UweTrottmann** [comment]
Sorry for taking so long. I used annotations and a simple check to get rid off the nullability warnings.

Merged from custom branch. This will be in the next update.

Thanks again!

**Pull request closed without merging**